server/plugin.go

```diff
@@ -119,6 +119,15 @@ func (p *Plugin) OnActivate() error {
 		p.singletonLockAcquired = true
 	}
 
+	// The metrics plugin is dependent on the metrics endpoint being expoesed, so we need to ensure it is enabled.
+	if cfg := p.API.GetUnsanitizedConfig(); cfg.MetricsSettings.Enable == nil || !*cfg.MetricsSettings.Enable {
```

Comment:
Just checking, what happens if a user downloads and installs this plugin on a TE build?

Reply:
Good call, it will enable performance monitoring. But assuming there won't be an `/metrics` endpoint so it will fail to scrape anyway.

Reply:
Wondering if we can atleast do a license check for enterprise before enabling this? Just as a sanity check.

Reply:
Sure